subsys/pm/policy.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pm/pm.h>
+#include <pm/policy.h>
+#include <spinlock.h>
+#include <sys_clock.h>
+#include <sys/__assert.h>
+#include <sys/time_units.h>
+#include <sys/atomic.h>
+
+/** State disallow reference counting */
+static atomic_t state_disallow_cnt[PM_STATE_COUNT];
+/** Callback to notify when state allowed status changes */
+static pm_policy_state_allowed_changed_cb_t allowed_changed_cb;
```

Comment:
What if several modules have to be notified about policy/latency change?
It looks that we can have only one registered callback, but exported API is supposed to be used by multiple clients.

Reply:
This is intended for SoC-specific usage (that's the requirement I got). If we want to have a list of clients, we need to clarify the use case.